### PR TITLE
 Closes #266 | Create dataset loader for SIB-200

### LINKED ISSUE #266: Create dataset loader for SIB-200
Dataloader name: `sib200/sib200.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?sib200

|     Dataset| sib200 |
|-------------|---|
| Description | SIB-200 is a topic classification data set covering 200 languages and derived from Flores-200 machine translation corpus. The English subset of Flores-200 was annotated with topic labels and those labels were projected onto the parallel instances belonging to the remaining 203 languages covered in the corpus. In initial baseline experiments on SIB-200, languages unseen during pre-training of multilingual LMs, under-represented language families, and languages from regions of Africa, Americas, Oceania, and SEA often had the lowest performance. Note that, after annotation, topic categories with less than 80 sentences were removed from the final classification dataset. |
| Subsets | ace, ban, bjn, bug, ceb, ilo, ind, jav, kac, khm, lao, lus, min, mya, pag, shn, sun, tgl, tha, vie, war, zsm |
| Languages | ace, ban, bjn, bug, ceb, ilo, ind, jav, kac, khm, lao, lus, min, mya, pag, shn, sun, tgl, tha, vie, war, zsm |
| Tasks | Text Classification |
| License | Apache license 2.0 (apache-2.0) |
| Homepage | https://github.com/dadelani/sib-200 |
| HF URL | - |
| Paper URL | https://arxiv.org/pdf/2309.07445v1.pdf |

### PR DESCRIPTION
**Pull request by tellarin**
Closes #266 

### Checkbox
- [ ] Confirm that this PR is linked to the dataset issue.
- [ ] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [ ] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [ ] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [ ] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [ ] Confirm dataloader script works with `datasets.load_dataset` function.
- [ ] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/sib_200/sib_200.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+_DATASETNAME = "sib_200"
+
+_DESCRIPTION = """\
+SIB-200 is the largest publicly available topic classification dataset based on Flores-200 covering 205 languages and dialects.
+The train/validation/test sets are available for all the 205 languages.
+"""
+
+_HOMEPAGE = "https://github.com/dadelani/sib-200"
+
+_LANGUAGES = [
```
Comment (holylovenia) [inline seacrowd/sea_datasets/sib_200/sib_200.py:49]: Please only use the list of language IDs for `_LANGUAGES` for consistency, but the subset names should retain the transcript types as well.

**Review by holylovenia — CHANGES_REQUESTED** [review]
Hi @tellarin, thanks for your dataloader.

I find the subsets do not work though. I tried using the `sib_200_ace_Latn_seacrowd_text` and `sib_200_ind_Latn_seacrowd_text`, but both of them return this output, which is unexpected:
```
DatasetDict({
    train: Dataset({
        features: ['id', 'text', 'label'],
        num_rows: 143705
    })
    validation: Dataset({
        features: ['id', 'text', 'label'],
        num_rows: 20295
    })
    test: Dataset({
        features: ['id', 'text', 'label'],
        num_rows: 41820
    })
})

lol
Train 0
{'id': '431',
 'label': 0,
 'text': 'توركي د كليلڠ لى لاؤة بك تلو لاݢي: لاؤة ايݢيا د بارات, ڠن لاؤة ايتم '
         'د اوتارا, ڠن لاؤة ميديتيرانيا د سلاتن.'}

Train 10
{'id': '1439',
 'label': 5,
 'text': 'هنا تريب عوه تلس بيچه ڤراڠ ڽن, اورڠ بريتانيا ݢڤڤهون ڤتهن رج اوه د '
         'لاؤة ك اورڠ جرمان.'}

Train 20
{'id': '555',
 'label': 6,
 'text': 'اورڠ اتليت موڠكين ݢراس ڤرلو ك دابس ادت ڤيه مڠڽو دابه ڽن هنا مساڤوى '
         'نا منفعت ڽڠ ڽات.'}

Train 30
{'id': '1441',
 'label': 5,
 'text': 'اورڠ بريتاين ݢجاݢ ولايه لاؤة لى بڠسا ك ڤتهن مندون كاڤل نك بيك جد جي '
         'تاموڠ لم ولايه لاؤة ڽڠ دوم ڤت نا, ڽن جد ك سابب چلاک بهڤيه كاڤل '
         'بيئسا.'}

Train 40
{'id': '1941',
 'label': 3,
 'text': 'جالن رايك ڽڠ لواس, رومه بينتيه كاچ ڠن ڤوست بلنجا موديرن ڽڠ دڤڠوى دڠن '
         'بوبوڠ ميراه , ڤاسي ابد ك-١٨, مسجيد توها ڠن ݢريجا , ادت ڤيه كوتا ڽي '
         'كأداءن جيه لبيه تراس ايروڤا ميديترانين درڤد متوركي.'}

Train 50
{'id': '1160',
 'label': 5,
 'text': 'اوه لهوه ڤتوݢس ݢتيليق لوم ايدينتيتس ڤميليه، ڤميليه ݢڤتاموڠ امڤلوڤ او '
         'لم ڤلوق سو ڠون ݢتيکين جادوي ڤميليه.'}

Train 59
{'id': '505',
 'label': 1,
 'text': 'نا اسڤيك ڤوسيتيف داري سيتوس ويب, ڽڠ مليڤوتي, جد دڠن موده ݢاتور لامن '
         'ݢله ڽڠ جد مچاكوڤ بلوݢ, فيديو, فوتو, ڠن فيتور لاءين جيه.'}

Train 60
{'id': '1333',
 'label': 1,
 'text': 'کاڤاي تربڠ ڽو ڤکى تکنولوݢي برباسيس رادر دارت ڽڠ لوبيه توها اونتوق '
         'ݢڤموڠکن کى کونترول لالو لينتس اودارا ݢڤتنتى کاڤاي تربڠ دڠن ڤريسيسي '
         'لوبيه رايق ڠن بري اينفورماسي ڽڠ لوبيه اکورت کى ڤيلوت.'}

Train 70
{'id': '679',
 'label': 0,
 'text': 'لى ترجون فيكتوريا ڽنكه بك سابوه زيمببوي بله بارت, لڤس چر نيبك '
         'ليفيڠستون, زمبيا, ڠن تي بوستون.'}
```

Do you also encounter the same behavior, @SamuelCahyawijaya?
You might want to check the dataloader with the unit test (e.g., `python -m tests.test_seacrowd seacrowd/sea_datasets/sib_200/sib_200.py` or `python -m tests.test_seacrowd seacrowd/sea_datasets/sib_200/sib_200.py --subset_id=sib_200_ace_Latn`) to make sure everything works well, @tellarin.

Also, I have some suggestions below.

**Comment by SamuelCahyawijaya** [comment]
Hi @tellarin, is there any problem with updating the PR? Please let us know if there is any difficulty  👍🏻

**Comment by tellarin** [comment]
> Hi @tellarin, is there any problem with updating the PR? Please let us know if there is any difficulty 👍🏻

I should be able to push the changes in the next couple days. Thanks.

**Comment by holylovenia** [comment]
> > Hi @tellarin, is there any problem with updating the PR? Please let us know if there is any difficulty 👍🏻
> 
> I should be able to push the changes in the next couple days. Thanks.

Sure, please let us know if you need any help with the dataloader. 🙏

**Comment by holylovenia** [comment]
Hi @tellarin, we from SEACrowd are currently looking forward to using your dataloader in our experiments. Do you think it will be possible for you to address the questions and/or requested changes? 😂 We aim to merge it by the end of this week.

**Comment by tellarin** [comment]
Opening the other PRs and fixing this ASAP.

**Comment by holylovenia** [comment]
> Opening the other PRs and fixing this ASAP.

Thanks @tellarin! Will be looking forward to them. ❤️

**Comment by SamuelCahyawijaya** [comment]
@tellarin : Any update on this PR?  if you are not available at the moment, I can help updating the script. 
For the point calculation we can split among the two of us. What do you think?

**Comment by holylovenia** [comment]
Hi @tellarin, thanks for your contribution up until now. Just a heads up, I would like to let @SamuelCahyawijaya work on this PR if there's no update until Tuesday, 16 April 2024 EoD AoE (23:59 UTC-12).

File: seacrowd/sea_datasets/sib_200/sib_200.py (hunk 1 above)
Comment (SamuelCahyawijaya) [reply seacrowd/sea_datasets/sib_200/sib_200.py:49]: Done

**Review by holylovenia — APPROVED** [review]
LGTM!! Thanks a lot, @SamuelCahyawijaya and @tellarin!! 🙏 Merging now.
